### ISSUE TITLE
`systemd` upgrade failure

### ISSUE THREAD

**Issue by elsaco** [microsoft/WSL#8871]
### Version

10.0.22621.521

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.62.1-microsoft-standard-WSL2

### Distro Version

Ubuntu 22.04.1

### Repro Steps

run `apt update` and check if new version of `systemd` is available
run `apt upgrade systemd`

### Expected Behavior

successful package upgrade

### Actual Behavior

Output on Ubuntu:

```
Unpacking systemd (249.11-0ubuntu3.6) over (249.11-0ubuntu3.4) ...
dpkg: error processing archive /tmp/apt-dpkg-install-eZFUq7/3-systemd_249.11-0ubuntu3.6_amd64.deb (--unpack):
 unable to make backup link of './usr/lib/tmpfiles.d/tmp.conf' before installing new version: Invalid cross-device link
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
dpkg: considering deconfiguration of systemd, which would be broken by installation of udev ...
dpkg: yes, will deconfigure systemd (broken by udev)
```

Output on Debian:

```
Unpacking systemd (247.3-7+deb11u1) over (247.3-7) ...
dpkg: error processing archive /var/cache/apt/archives/systemd_247.3-7+deb11u1_amd64.deb (--unpack):
 unable to make backup link of './usr/lib/tmpfiles.d/tmp.conf' before installing new version: Invalid cross-device link
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/systemd_247.3-7+deb11u1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

```

**Comment by elsaco** [comment]
Quick workaround is to start w/out systemd, upgrade, then re-enable `systemd = true` in `/etc/wsl.conf`. Tested on Ubuntu.

**Comment by Biswa96** [comment]
Duplicate of https://github.com/microsoft/WSL/issues/8847

**Comment by benhillis** [comment]
/dupe #8847 

Should be resolved with 0.68.2
